### PR TITLE
fix(downloader): only delete temp file if it exists

### LINKED ISSUE #5: npm test shows error, returns exit code 0
Output

```
➜  fxa-geodb git:(TDA-master) npm test

> fxa-geodb@0.0.3 test /Users/vladikoff/mozilla/fxa-geodb
> mocha



  fxa-geodb
    ✓ returns a promise when called
    ✓ returns an error object with `UNABLE_TO_OPEN_FILE` when supplied with an non-existent file
    ✓ returns an error object with `IS_INVALID` when supplied with an undefined ip variable
    ✓ returns an error object with `IS_INVALID` when supplied with an object
    ✓ returns an error object with `IS_INVALID` when supplied with an empty ip
    ✓ returns an error object with `IS_INVALID` when supplied with an invalid ip
    ✓ returns an object with location data when supplied with a valid ip address
    ✓ returns an Error Object when no data is available
    ✓ returns an object with partial location data when complete data is not available

  maxmind-db-downloader
    createTargetDir
      ✓ creates the specified directory
      ✓ does nothing when directory already exists
    setupDownloadList
      ✓ sets up the download list from sources.json
    downloadAll
      ✓ calls Promise.all with a promise array (71ms)
    setupAutoUpdate
Unhandled rejection Error: ENOENT: no such file or directory, unlink '/Users/vladikoff/mozilla/fxa-geodb/test-db/cities-db.mmdb-temp'
    at Error (native)
    at Object.fs.unlinkSync (fs.js:932:18)
    at /Users/vladikoff/mozilla/fxa-geodb/lib/maxmind-db-downloader.js:94:20
    at tryCatcher (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/promise.js:504:31)
    at Promise._settlePromise (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/promise.js:561:18)
    at Promise._settlePromiseCtx (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/promise.js:598:10)
    at Async._drainQueue (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/async.js:143:12)
    at Async._drainQueues (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/async.js:148:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/vladikoff/mozilla/fxa-geodb/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate [as _immediateCallback] (timers.js:383:17)

      ✓ auto update calls downloadAll correctly (5021ms)


  14 passing (6s)

➜  fxa-geodb git:(TDA-master) 
```

Thing failed, but returned 0. is it supposed to be that way @TDA?:

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #5

### CODE REVIEW

**Comment by vladikoff** [comment]
@vbudhram r?

File: lib/maxmind-db-downloader.js [hunk 1]
```diff
@@ -95,7 +95,9 @@ var MaxmindDbDownloader = function () {
               }, function (err) {
                 // download resulted in an error, do not rename
                 // remove temp file
-                fs.unlinkSync(targetFilePathTemp);
+                if (fs.existsSync(targetFilePathTemp)) {
```
Comment (vbudhram) [inline lib/maxmind-db-downloader.js]: I believe `fs.existsSync` is being [deprecated](https://nodejs.org/api/fs.html#fs_fs_existssync_path). Could use `fs.statSync`, and catch error if file doesn't exists.

Comment (vladikoff) [inline lib/maxmind-db-downloader.js]: Let's wait to see what happens in node land. I'm guessing this will be a larger problem with node 8

Comment (vladikoff) [inline lib/maxmind-db-downloader.js]: nice find too 👍

**Pull request merged**